website/content/en/docs/best-practices/observability-best-practices.md

```diff
@@ -65,6 +65,27 @@ The most common types are:
  - `Gauge` Value can be increased and decreased as needed.
 5. `_total` suffix should be used for accumulating count. If your metrics has labels with high cardinality, like `pod`/`container` it usually means that you can aggregate it more, thus it will not require `_total` suffix.
 
+#### Recommended Metrics
+
+* `<operator-name-prefix>_operaor_health_status` - The metric reports the operator health status. It should be reported by the operator itself based on the internal operator validations.  
+  This metric allows the cluster users, to view all operators health statuses with a single metric and improves the overall cluster observability.
+
+  In order to maintain consistency, it is recommended to use the following health statuses as the metric values:
+  * `0` - Indicates that the operator is healthy and working as expected.
+  * `1` - Indicates that the operator has some issues that needs to be addressed and can potentially lead to loss of functionality.
+  * `2` - Indicates that the operator is unhealthy and there is a loss of functionality that should be addressed.
+
+  Metric labels:
+  * `name` - The operator name.[Recommended].
+  * `reason` - The reason for the health issue, if possible.[Optional].
+
+  **Query:** To view all operators health statuses with a single query, users can use the following PromQL query, `{__name__=~".*operator_health_status"}`.
+
+  **Note:** In order for this metric to be useful for the users, it MUST be reported in a consistent way across all the operators. If you would prefer to follow any other level of granularity to indicate health statuses, it should be clearly documented such that an external observer can infer the indicator values accordingly.  
+
+  It is suggested to report the metric values as numbers that represent the health statuses, since using strings like: Healthy/Warning/Error, can be implemented in many ways that would cause consistency issues and disagreements, like terminology preferences (For example: `Error` vs. `Unhealthy`), spelling errors, capital letters etc.
```

Comment:
This is the kind of thing I was thinking of for why a library could go a long ways here. The library owns the way the metric gets set. Users of the library (i.e. operator authors) would not even need to be concerned at all with what the metric ended up looking like and they may not even need to be concerned with the way we expose that information. In the future, we might also think its a good idea to emit an event. Operator authors would just say "make sure the thing that exposes my operator health is updated to say I am healthy/degraded/failing/unknown/whatever" and the library is the thing that makes sure all operators do whatever stuff on the backend consistently.

Reply:
@joelanford  At the moment I believe this is a good start, since the library will still be based on this best practice.
We plan to add a monitoring plugin where we can add this as well as part of that effort.